interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is the syntax language used in react that allows for the use of JavaScript logic within a React app html file. The main syntax difference between HTML and JSX being that JSX requires the use of curly braces {}. 
 
-Researched answer:
+Researched answer: The difference between JSX and HTML is that JSX uses the syntax className where HTML uses class. JSX is what allows React to create user interfaces by interacting with the different React components.
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-Your answer:
+Your answer: Yarn stands for Yet Another Resource Negotiator and is a package manager that was created by Facebook in 2016. It stores the different dependencies or files needed to run jest tests. Yarn modifies the React application by adding two files: package.json, and yarn.lock.
 
-Researched answer:
+Researched answer: Yarn is necessary to create a React app due to the fact that a React Application is made of thousands of individual code snippets dependencies called node modules. The node modules must be organized in a particular order, and yarn ensures that they are correctly place for the framework of a React app.
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function is a function that does not have a specified parameter for what type of argument can be passed through. This allows multiple types of data to be passed through a single function for continuous iteration for different scenarios. 
 
-Researched answer:
+Researched answer: An anonymous function is a function that does not have any declared name for the function. Instead of using a function name to declare a function, ie. const functionName = () => {}, we can instead use function = () => {}. The function will only be called by a variable it is stored in as a function as a value. 
```

Comment:
Storing an anonymous function to a variable would then be giving it a name.  More commonly, you will see an anonymous function passed as an argument where it can only be called on at that moment and nowhere else in your code.